packages/fab/index.tsx

```diff
@@ -0,0 +1,79 @@
+// The MIT License
+//
+// Copyright (c) 2018 Google, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+import * as React from 'react';
+import * as classnames from 'classnames';
+import withRipple from '@material/react-ripple';
+
+export interface FabProps {
+  mini: boolean,
+  icon?: React.ReactElement<HTMLElement>,
+  textLabel: string,
+  className: string,
+  initRipple: (surface: HTMLButtonElement) => void,
+  unbounded: boolean,
+};
+
+const Icon: React.FunctionComponent<{icon?: React.ReactElement<HTMLElement>}> = ({icon}) => {
+  if (!icon) {
+    return null;
+  }
+  const updatedProps = {
+    className: classnames('mdc-fab__icon', icon.props.className),
+  };
+  return React.cloneElement(icon, updatedProps);
+};
+
+const TextLabel: React.FunctionComponent<{textLabel: string}> = ({
+  textLabel, // eslint-disable-line react/prop-types
```

Comment:
Are the eslint comments a temporary measure until we completely convert everything? 

We may want to make another issue to install [google/ts-tyle](https://github.com/google/ts-style) into this repo.

Reply:
ya temporary. I didn't want to break all the other js files until all are converted to TS

Reply:
https://github.com/material-components/material-components-web-react/issues/507